src/lib/shaders/scene-fragment-shader.ts

```diff
@@ -375,17 +383,59 @@ void main () {
   float targetDistance = getDist(targetSides, varyingTargetShapeTexture);
   float signedDistance = mix(previousDistance, targetDistance, varyingT);
 
-  vec4 color =
-    signedDistance < varyingBorderThresholds.x ? vec4(0.) :
-    signedDistance < varyingBorderThresholds.y ? varyingBorderColor :
-    varyingFillColor;
+  // Create an antialiasing window around the determined signed distance with
+  // radius equal to 1 device pixel (diameter of 2 device pixels).
+  vec2 window = signedDistance +
+    varyingEdgeToPixelScale * antialiasingFactor * vec2(-1., 1.);
+
+  // Width of the antialiasing window.
+  float width = window.y - window.x;
+
+  // Determine the contribution to the window of the border and fill.
+  vec2 contrib;
+
+  if (width > 0.) {
+    // Amount of space within the window that overlaps the border.
+    contrib.x =
+      min(varyingBorderThresholds.y, window.y) -
+      max(varyingBorderThresholds.x, window.x);
+
+    // Amount of space within the window that overlaps the fill color. May be
+    // negative, if no part of the window overlaps.
+    contrib.y = width - (varyingBorderThresholds.y - window.x);
+
+    // Normalize contributions to the antialiasing window's width and clamp to
+    // possible contribution range.
```

Comment:
nit: move clamp comment down to line 418 so it's correctly localized

Reply:
Done.